### PR TITLE
split wasm filter

### PR DESCRIPTION
**Pull request by adilhafeez**
- split llm and prompt wasm filter out
- move http and stats modules to public_types
- update tests, checks and linters on github actions
- update dockerfile and envoy.template.yaml file to use separate wasm filters

This is part 1 of two part code change. In next PR I will remove unnecessary code from both llm and prompt gateway code.

fixes https://github.com/katanemo/arch/issues/188

### CODE REVIEW

File: crates/prompt_gateway/src/ratelimit.rs
Comment (junr03) [inline crates/prompt_gateway/src/ratelimit.rs:1]: Why is rate limit not part of the common code?

Comment (junr03) [inline crates/prompt_gateway/src/ratelimit.rs:1]: Why is rate limit not part of the common code?

File: crates/prompt_gateway/src/tokenizer.rs
Comment (junr03) [inline crates/prompt_gateway/src/tokenizer.rs:1]: Why is the tokenizer not part of the common code?

File: crates/public_types/src/common_types.rs
Comment (junr03) [inline crates/public_types/src/common_types.rs:1]: I am thinking that the `public_types` crate should be `common`, and potentially common_types needs to be separated into different modules.

**Review by junr03 — COMMENTED** [review]
Good first step.

File: crates/public_types/src/common_types.rs
Comment (adilhafeez) [reply crates/public_types/src/common_types.rs:1]: I was hoping to do that in the next PR so this PR is smaller and simpler. But sure let me take care of it now.

File: crates/prompt_gateway/src/ratelimit.rs
Comment (adilhafeez) [reply crates/prompt_gateway/src/ratelimit.rs:1]: moved

File: crates/prompt_gateway/src/tokenizer.rs
Comment (adilhafeez) [reply crates/prompt_gateway/src/tokenizer.rs:1]: moved

**Review by junr03 — APPROVED** [review]

**Pull request merged**